tests/rspec/lib/cluster.rb

```
@@ -141,6 +147,35 @@ def wait_til_ready
     end
   end
 
+  def wait_nodes_ready
+    puts 'Waiting for nodes become in ready state after an update'
+    from = Time.now
+    loop do
+      begin
+        nodes = KubeCTL.run_and_parse(@kubeconfig, 'get nodes')
+        nodes_ready = Array.new(@tfvars_file.node_count, false)
+        nodes['items'].each_with_index do |item, index|
+          item['status']['conditions'].each do |condition|
+            if condition['type'] == 'Ready' && condition['status'] == 'True'
+              nodes_ready[index] = true
+            end
+          end
+        end
+
+        break if nodes_ready.uniq.length == 1
+
+        puts 'One or more nodes are not ready yet or missing nodes. Waiting...'
+        puts "# of returned nodes #{nodes['items'].size}. Expected #{@tfvars_file.node_count}"
+        raise StandardError
```

Raising an error without an error message seems not to be a good practice to me. Can you either throw a custom error message (See https://github.com/coreos/tectonic-installer/blob/master/tests/rspec/lib/kubectl_helpers.rb#L11), or add an error message to the `StandardError`?

---

How about making the above two `puts` statements part of the error message instead?

---

ok, will refactor